spectral_cube/tests/test_masks.py

```diff
@@ -256,6 +256,30 @@ def test_wcs_validity_check():
     # just checking that this works, not that it does anything in particular
     moment_map = s3.moment(order=1)
 
+def test_wcs_validity_check_failure():
+    cube, data = cube_and_raw('adv.fits')
+    wcs2 = cube.wcs.copy()
+    # add some difference in the 4th decimal place
+    wcs2.wcs.crval[2] += 0.00001
+
+    # can make a mask
+    mask = BooleanArrayMask(data>0, cube._wcs)
+
+    # but if it's not exactly equal, an error should be raised at this step
+    with pytest.raises(ValueError) as exc:
+        cube = cube.with_mask(mask)
+    assert exc.value.args[0] == "WCS does not match mask WCS"
+
+    # this one should work though
+    cube = cube.with_mask(mask, wcs_tolerance=1e-4)
+    assert cube._wcs_tolerance == 1e-4
```

Comment:
If a user does:

```
cube = cube.with_mask(mask, wcs_tolerance=1e-4)
cube = cube.with_mask(mask, wcs_tolerance=1e-6)
```

The internal state will end up different from:

```python
cube = cube.with_mask(mask, wcs_tolerance=1e-6)
cube = cube.with_mask(mask, wcs_tolerance=1e-4)
```

What about requiring the user to set wcs_tolerance as a settable property on the cube before adding masks?